### DISCUSSION TITLE
Popup when hovering or clicking a symbol

### DISCUSSION THREAD

**Discussion by Japhys** [nazka/mapbox-gl-js-spiderfy#23] [Q&A] [answered]
Hi,

Thanks for this plugin! I am still very new to this (maplibre) and not a developer so please bear with me :) I was wondering if it would be possible to display an infobox when clicking on a symbol?
I tried to use this example to no avail:
https://maplibre.org/maplibre-gl-js-docs/example/popup-on-hover/
Is it at all possible?
Kind regards

**Comment by simondriesen** [comment, marked as answer]
Hello,

For clicking you can use the `onLeafClick` option but let me know if your having any issues. I didn't add an `onLeafHover` option but this does sound useful so I will add it soon.

```js
const popup = new maplibregl.Popup();

const spiderfy = new Spiderfy(map, {
  onLeafClick: (f) => {
    // set your popup (clicked feature data is inside `f` object)
    const coordinates = f.geometry.coordinates.slice();
    const description = f.properties.description;
    popup.setLngLat(coordinates).setHTML(description).addTo(map);
  },
   ...
})
```

Note: You probably want to put `closeOnLeafClick` option to false if you want to be able to click on the leaves without them disappearing. But this means you have to make your own closing logic by manually closing them using `spiderfy.unspiderfyAll()`

**Comment by Japhys** [comment]
Wonderful! I will give it a try. And indeed, it would be cool if you could add hover too. Thanks so much.

**Comment by simondriesen** [comment]
Just added the `onLeafHover` option, so feel free to get the latest version and try it out. Thanks for the feedback!
